src/app/shell/header/header/header.component.html

```diff
@@ -1,11 +1,12 @@
-<ng-container [ngSwitch]="headerType$ | async">
-  <ish-header-error *ngSwitchCase="'error'" />
-  <ish-header-simple *ngSwitchCase="'simple'" />
-  <ish-header-checkout *ngSwitchCase="'checkout'" />
-
-  <ng-container *ngSwitchDefault>
-    <ish-header-default [isSticky]="isSticky$ | async" [deviceType]="deviceType$ | async" [reset]="reset$ | async" />
-  </ng-container>
+<!-- using ngIf instead of ngSwitch to resolve an issue with hydration and duplicate headers for special logins -->
+<ng-container *ngIf="headerType$ | async as headerType; else defaultHeader">
```

Comment:
in case the headerType is anything else than simple, error or checkout you don't show the default header now. You could map the headerType to an empty string in the ngOnInit event
 ngOnInit() {
    this.headerType$ = this.appFacade.headerType$.pipe(map type -> ...)

Reply:
added headerTypes and the check for a valid header type , otherwise default header is used